### ISSUE TITLE
More questions about using the library without define-application

### ISSUE THREAD

**Issue by bigos** [bohonghuang/cl-gtk4#12]
This is related to https://github.com/bohonghuang/cl-gtk4/issues/11 , where I encountered a surprise. I was assuming I can completely remove define-application. While trying this I got this error:

> Cannot expand DEFINE-MAIN-WINDOW outside DEFINE-APPLICATION.

I am exploring this path because there are some programs that use several windows. One such example would be a paint program with a separate toolbar window. While you perform painting on canvas window, and it could be considered to be the main window, there are situations with several windows and no one of them really being the main window. Some text editors like Emacs could be such examples. When I have multiple windows open and I close the first window, what happens with the concept of the main window? Would the remaining windows still be open and fully functional?

**Comment by bigos** [comment]
I found it.
https://github.com/bigos/Pyrulis/blob/3b7eedd20e5d287509261f526c71ba09058fe438/Lisp/menu-cairo-test2.lisp#L182

This way, I can remove the `define-application`  and move ahead.

**Comment by bigos** [comment]
The case is closed for me. My questions started because I could not run this:
https://github.com/bigos/Pyrulis/blob/57cd95341ffdb9f7418e5da4621aab87d23be611/Lisp/cl-gtk4-tictactoe.lisp#L679
after the update of your libraries.

I guess I have reached the stage where I can modify your examples to suit my needs.

**Comment by bohonghuang** [comment]
> Some text editors like Emacs could be such examples. When I have multiple windows open and I close the first window, what happens with the concept of the main window?

Some frameworks, such as [relm4](https://relm4.org/), have this concept, and so does the `define-application` in `cl-gtk4`, which is suitable for most cases, except in some situations where the programming can create windows dynamically. In those cases, `cl-gtk4` cannot identify and reconstruct the windows for you while the application runs, because how and when to construct these windows is controlled by the programmer, but a main window doesn't need such a context. You can certainly avoid the usage of `define-application` just as I did in examples before the commit 7cba4f17f250ae38755ef4bd592e4a78645999a8 for those cases.

**Comment by bohonghuang** [comment]
> Would the remaining windows still be open and fully functional?

Yes. The GTK application exits automatically only if all the windows are closed.

**Comment by bigos** [comment]
One day I may create an example and come back to you with more questions. But for now, it is everything I wanted to know. 

Thank you very much for your help.
